plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,374 @@
+		SNRSignalStrength  string `json:"SNR Signal Strength"`
+		RSRPSignalStrength string `json:"RSRP Signal Strength"`
+		RegistrationStatus string `json:"Registration Status"`
+		ActiveBandClass    string `json:"Active Band Class"`
+		Carrier            string `json:"Carrier"`
+		ConnectionStatus   string `json:"Connection Status"`
+		Failure            struct {
+			Reason string `json:"reason"`
+		} `json:"failure"`
+	} `json:"LTE"`
+}
+
+type lteDeviceInfo struct {
+	DeviceInfo struct {
+		IMEI  string `json:"imei"`
+		ICCID string `json:"iccid"`
+		IMSI  string `json:"imsi"`
+	} `json:"Device Info"`
+}
+
+// T128Lte is an input for metrics of a 128T router instance
+type T128Lte struct {
+	CollectorName  string `toml:"collector_name"`
+	routerName     string
+	nodeName       string
+	t128ConfigFile string
+	ltePathDir     string
+	globalInitPath string
+	localInitPath  string
+}
+
+// SampleConfig returns the default configuration of the Input
+func (*T128Lte) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*T128Lte) Description() string {
+	return "Make a 128T LTE query and return the data"
+}
+
+func (plugin *T128Lte) Init() error {
+	var err error
+	plugin.routerName, err = plugin.loadRouterName()
+	if err != nil {
+		return err
+	}
+	plugin.nodeName, err = plugin.loadNodeName()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (plugin *T128Lte) Gather(acc telegraf.Accumulator) error {
+	foundAnyLTEInterface := false
+	for _, lteInterface := range plugin.getLteInterfaces() {
+		foundAnyLTEInterface = true
+		lteMetric, err := plugin.getLteStateInfo(lteInterface, plugin.ltePathDir)
+		if !hasRequiredData(lteMetric) {
+			acc.AddError(fmt.Errorf("Interface %s does not include required data to be included in the collector", lteInterface.Name))
+			continue
+		}
+		lteInfo, err := plugin.getLteInfo(lteInterface, plugin.ltePathDir)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to find LTE info %w", err))
+			continue
+		}
+		tags := map[string]string{
+			"device-interface": lteInterface.Name,
+		}
+		fields := map[string]interface{}{
+			"signal-strength":   lteMetric.LTE.SignalStrength,
+			"carrier":           lteMetric.LTE.Carrier,
+			"connection-status": lteMetric.LTE.ConnectionStatus,
+			"active-band-class": lteMetric.LTE.ActiveBandClass,
+			"apn":               lteMetric.LTE.Active3GPPProfile.APN,
+			"service-mode":      lteMetric.LTE.RadioInterface,
+			"service-status":    getServiceStatus(lteMetric),
+			"imsi":              lteInfo.DeviceInfo.IMSI,
+			"imei":              lteInfo.DeviceInfo.IMEI,
+			"iccid":             lteInfo.DeviceInfo.ICCID,
+		}
+		signalValue := getSignalValue(lteMetric)
+		for k, v := range signalValue {
+			fields[k] = v
+		}
+		//one last check to make sure the fields don't exist if values are empty
+		for fieldName, value := range fields {
+			if value == "" {
+				delete(fields, fieldName)
```

Comment:
not sure if it makes sense to delete those fields or add another check of checking if the value is "N/A", @agrawalkaushik @gregschrock @chao-128t any insights?

Reply:
based on the python script, I believe fields were excluded when empty